File: src/Application Specific/Classes/Entity Collections/ContractOriginalSiteRates/ContractOriginalSiteRatesValidator.vb
Namespace Entity

Namespace ContractOriginalSiteRatess

Public Class ContractOriginalSiteRatesValidator
    'make sure that contact object is valid
    Inherits BaseValidator

    Public Sub Validate(ByVal oContractOriginalSiteRates As ContractOriginalSiteRates)

        'make sure that contact object is valid
        If oContractOriginalSiteRates.Errors.Count > 0 Then
            Dim de As DictionaryEntry
            For Each de In oContractOriginalSiteRates.Errors
                Me.AddCriticalMessage(CStr(de.Value))
            Next
        End If

        
	
        If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
            Throw New ValidationException(Me)
        End If

    End Sub

End Class

End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/ContractOriginalSiteRates/ContractOriginalSiteRatesValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace ContractOriginalSiteRatess
    {
        // make sure that contact object is valid
        public class ContractOriginalSiteRatesValidator : BaseValidator
        {
            public void Validate(ContractOriginalSiteRates oContractOriginalSiteRates)
            {

                // make sure that contact object is valid
                if (oContractOriginalSiteRates.Errors.Count > 0)
                {
                    DictionaryEntry de;
                    foreach (var de in oContractOriginalSiteRates.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}